### PR TITLE
Assignment5

### PR DESCRIPTION
**Pull request by pnotequalnp**
WIP

### CODE REVIEW

**Comment by pnotequalnp** [comment]
Some commits are out of order because of system clock troubles...

File: Mailbox/DataLoader.cs [hunk 1]
```diff
@@ -1,21 +1,58 @@
...
+            if (mailboxes is null) throw new ArgumentNullException(nameof(mailboxes));
+
+            Source.Position = 0;
+            using var writer = new StreamWriter(Source, leaveOpen: true);
+            writer.WriteLine(JsonConvert.SerializeObject(mailboxes));
         }
 
-        public void Save(List<Mailbox> mailboxes)
+        #region IDisposable Support
+        private bool disposedValue = false;
```
Comment (Keboo) [inline Mailbox/DataLoader.cs]: Fields should be specified in _PascalCase, or switched to use an auto property

File: Mailbox/Program.cs [hunk 2]
```diff
@@ -43,9 +44,9 @@ static void Main(string[] args)
                         Console.WriteLine("Enter the last name");
                         string lastName = Console.ReadLine();
                         Console.WriteLine("What size?");
-                        if (!Enum.TryParse(Console.ReadLine(), out Size size))
+                        if (!Enum.TryParse(Console.ReadLine(), out Sizes size))
                         {
```
Comment (Keboo) [inline Mailbox/Program.cs]: Parameter args of method Main is never used. Remove the parameter or use it in the method body.

File: Mailbox/Person.cs [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+        {
+            FirstName = firstName ?? throw new ArgumentNullException(nameof(firstName));
+            LastName = lastName ?? throw new ArgumentNullException(nameof(lastName)); ;
+        }
+
+        public override bool Equals(object? obj) =>
+            obj is Person other && Equals(other);
+
+        public bool Equals(Person other) =>
+            (FirstName, LastName) == (other.FirstName, other.LastName);
```
Comment (Keboo) [inline Mailbox/Person.cs]: Nice use of tuples for the comparison

File: Mailbox/Sizes.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        Medium    = 0b0010,
+        Large     = 0b0100,
+        Premium   = 0b1000,
+    }
+
+    public static class SizesExtensions
+    {
+        public static Sizes Size(this Sizes sizes) => sizes & ~Sizes.Premium;
+
+        public static bool Premium(this Sizes sizes) => sizes.HasFlag(Sizes.Premium);
```
Comment (Keboo) [inline Mailbox/Sizes.cs]: Similar to the coding standard for properties, you typically would do the same for methods.
`DO favor prefixing Boolean properties with “Is,” “Can,” or “Has,” when that practice adds value.`

```suggestion
        public static bool IsPremium(this Sizes sizes) => sizes.HasFlag(Sizes.Premium);
```

**Pull request closed without merging**